common/flatpak-transaction.c

```diff
@@ -3084,6 +3089,45 @@ request_tokens_for_remote (FlatpakTransaction *self,
   return TRUE;
 }
 
+static gboolean
+order_authed_operations (FlatpakTransaction *self,
+                        GCancellable        *cancellable,
+                        GError             **error)
+{
+  FlatpakTransactionPrivate *priv = flatpak_transaction_get_instance_private (self);
+  GList *l;
+
+  for (l = priv->ops; l != NULL; l = l->next)
+    {
+      FlatpakTransactionOperation *op = l->data;
+      FlatpakTransactionOperation *authenticator_op;
+
+      gboolean maybe_needs_token = FALSE;
+
+      /* Ensure we sort related refs that require tokens below the authenticator too */
+      if (op->fail_if_op_fails != NULL)
+        if (op->fail_if_op_fails->token_type != 0)
+          maybe_needs_token = TRUE;
```

Comment:
Why do we need this? The related refs will be in the priv->ops list anyway, so they'll be sorted in order when we get to it anyway?

Reply:
In testing, I found that a `.Locale` package belonging to a package requiring auth was being sorted before the authenticator and hence failing as it also requires the same token. Perhaps some of the other sorting isn't behaving as you'd expect?

It may be worth noting that I've been testing with an authenticator in the `io.elementary` namespace and applications requiring tokens in the `com.github` namespace. I notice there is some alphabetical sorting in there as well as a greedy algorithm that sorts ops that don't depend on anything first. I'd imagine either one of those could put the `.Locale` package before the authenticator.

Reply:
However, if there is a better way of fixing this issue, I'm open to suggestions.